lib/galaxy/tools/data/__init__.py

```diff
@@ -506,18 +515,18 @@ def parse_column_spec(self, config_element):
 
     def extend_data_with(self, filename, errors=None):
         here = os.path.dirname(os.path.abspath(filename))
-        self.data.extend(self.parse_file_fields(open(filename), errors=errors, here=here))
+        self.data.extend(self.parse_file_fields(filename, errors=errors, here=here))
         if not self.allow_duplicate_entries:
             self._deduplicate_data()
 
-    def parse_file_fields(self, reader, errors=None, here="__HERE__"):
+    def parse_file_fields(self, filename, errors=None, here="__HERE__"):
         """
         Parse separated lines from file and return a list of tuples.
 
         TODO: Allow named access to fields using the column names.
         """
         separator_char = "<TAB>" if self.separator == "\t" else self.separator
-
+        reader = open(filename)
```

Comment:
Thank you for making the change! One last comment (the rest looks fine to me). The `reader` is never closed here. So, maybe `with open(filename) as reader :`?

Reply:
This like doubles the size of the PR. I have killed this `reader`, long live `fh`. 

Thanks @ic4f